documentation/projects/proposals/data_normalization/20240227-implementation_plan_catalog_data_cleaning.md

```diff
@@ -0,0 +1,198 @@
+approximately just over **20 hours** to complete, according to a inspection of
+latest executions. The cleaned data is only saved to the API database, which is
+replaced each time during the same data refresh, causing it to have to be
+repeated each time to make the _same_ corrections.
+
+This cleaning process was designed this way to speed the rows update up since
+the relevant part was to provide the correct data to users via the API. Most of
+the rows affected were added previous to the creation of the `MediaStore` class
+in the Catalog (possibly by the discontinued CommonCrawl ingestion) which is
+nowadays responsible for validating the provider data. However, it entails a
+problem of wasting resources both in time, which continues to increase, and in
+the machines (CPU) it uses, which could easily be avoided making the changes
+permanent by saving them in the upstream database.
+
+[img-data-refresh]:
+  https://github.com/WordPress/openverse-catalog/blob/main/DAGs.md#image_data_refresh
+
+## Expected Outcomes
+
+<!-- List any succinct expected products from this implementation plan. -->
+
+- The catalog database (upstream) preserves the cleaned data results of the
+  current Ingestion Server's cleaning steps
+- The image Data Refresh process is simplified by reducing the cleaning steps
+  time to nearly zero (and optionally removing them).
+
+## Step-by-step plan
+
+The cleaning functions that the Ingestion Server applies are already implemented
+in the Catalog in the `MediaStore` class: see its `_tag_blacklisted` method
+(which probably should be renamed) and the [url utilities][url_utils] file. The
+only part that it's not there and can't be ported is the filtering of
+low-confidence tags, since provider scripts don't save an "accuracy" by tag.
+
+With this the plan then starts in the Ingestion Server with the following steps:
+
+1. [Save TSV files of cleaned data to AWS S3](#save-tsv-files-of-cleaned-data-to-aws-s3)
+1. [Make and run a batched update DAG for one-time cleanup](#make-and-run-a-batched-update-dag-for-one-time-cleanup)
+1. [Run an image Data Refresh to confirm cleaning time is reduced](#run-an-image-data-refresh-to-confirm-cleaning-time-is-reduced)
+
+[url_utils]:
+  https://github.com/WordPress/openverse/blob/a930ee0f1f116bac77cf56d1fb0923989613df6d/catalog/dags/common/urls.py
+
+## Step details
+
+### Save TSV files of cleaned data to AWS S3
+
+In a previous exploration, it was set to store TSV files of the cleaned data in
+the form of `<identifier> <cleaned_field>`, which can be used later to perform
+the updates efficiently in the catalog DB, which only had indexes for the
+`identifier` field. These files are saved to the disk of the Ingestion Server
+EC2 instances, and worked fine for files with URL corrections since this type of
+fields is relatively short, but became a problem when trying to save tags, as
+the file turned too large and filled up the disk, causing problems to the data
+refresh execution.
```

Comment:
Noting that the files must have a timestamp (or some indication of the date saved) in their names to make sure that we don't overwrite them on the next refresh.

Reply:
It doesn't matter much if the data is overwritten I think 🤔 In theory, it will always be the same given the new data is ingested cleaned.